packages/containers/src/AppLoader/AppLoader.saga.js

```diff
@@ -0,0 +1,52 @@
+import { api } from '@talend/react-cmf';
+import { call, select, all } from 'redux-saga/effects';
+import { delay } from 'redux-saga';
+import invariant from 'invariant';
+
+export const ACTION_CREATORS = 'actionCreators';
+export const WAIT_FOR = 'waitFor';
+
+/**
+ * This function wait until a collection is here
+ * @param {string} collectionName the name of the collection
+ * @param {int} interval the interval to check again if the collection is there
+ */
+export function* waitFor(collectionName, interval = 10) {
+	// eslint-disable-next-line no-constant-condition
+	while (true) {
+		const collection = yield select(api.selectors.collections.get, collectionName);
+		if (collection !== undefined) {
+			break;
+		}
+		yield call(delay, interval);
+	}
+}
+
+/**
+ * This function handle a specific step
+ * @param {object} step a bootstrap step that could contain a actionCreator list or a waitList
+ */
+export function* handleStep(step) {
+	if (step[ACTION_CREATORS]) {
```

Comment:
On TDC, we have some additional fetaures like TAKE_ACTION. so we can update it to reuse this.